lib/get-transitive-data.js

```diff
@@ -209,80 +203,98 @@ export default function getTransitiveData ({
-
-  // seek forward to correct path
-  let curIdx = 0
-  while (curIdx++ < pathIdx) {
-    const nSegments = origin.data[offset++]
-    offset += nSegments * 3
+  return {
+    ...transitiveNetwork,
+    journeys: clusteredJourneys,
+    places
   }
+}
 
-  const nSegments = origin.data[offset++]
-
-  if (nSegments > MAX_PATH_SEGMENTS) {
-    // bail if we have a ridiculous path (clearly a data/code error), and complain loudly
-    log(`Too many path segments (${nSegments} > ${MAX_PATH_SEGMENTS}) in path ${pathIdx} to stop ${stop} from origin ${origin.x}, ${origin.y}, returning no path. This implies a bug, please raise cain about it`)
-    return null
-  }
+type PathCount = {
+  count: number,
+  path: PathDescriptor
+}
 
-  const path = []
-  for (let seg = 0; seg < nSegments; seg++) {
-    // not delta coded
-    // todo this is creating typed arrays which may be inefficient for such small values
-    path.push(slice(origin.data, offset, offset + 3))
-    offset += 3
+function countCommonPaths (
+  memo: Array<PathCount>,
+  path: PathDescriptor,
+  index: number
+): Array<PathCount> {
+  if (index === 0) return [{count: 1, path}]
+  if (memo[0].path[0] !== path[0] || memo[0].path[1] !== path[1]) {
+    return [{count: 1, path}, ...memo]
   }
+  memo[0].count++
+  return memo
+}
 
-  return path
+const byCount = (p1, p2) => p2.count - p1.count
+const byStopIdAndPathIndex = (p1, p2) => {
+  const v0 = p1[0] - p2[0] // sort first by stop ID
+  if (v0 !== 0) return v0
+  else return p1[1] - p2[1] // then by path index
 }
+const uniquePathIds = (p, idx, arr) =>
+  idx === 0 || p[0] !== arr[idx - 1][0] || p[1] !== arr[idx - 1][1]
 
-/** Perform heuristic filtering on paths. How does it work? Magic. */
-export function filterPaths ({
+/**
+ * Get the most common paths
+ */
+function getCommonPaths ({
+  getRouteIndexFromPatternId,
   log,
-  paths,
-  transitiveNetwork
-}) {
-  const inPathCount = paths.length
-
-  // function to convert pattern ID to route ID
-  const r = (pattern) => {
-    const pat = transitiveNetwork.patterns.find(p => p.pattern_id === String(pattern))
-    return transitiveNetwork.routes.findIndex(r => r.route_id === pat.route_id)
+  origin,
+  paths
+}: {
+  getRouteIndexFromPatternId: Function,
+  log: Function,
+  origin: Origin,
+  paths: Array<PathDescriptor>
+}): Array<StopPatternStops> {
+  paths.sort(byStopIdAndPathIndex)
+  const pathCounts = paths.reduce(countCommonPaths, [])
+  pathCounts.sort(byCount)
+
+  log(`${pathCounts.length} unique paths`)
+  if (pathCounts.length > MAX_TRANSITIVE_PATHS) {
+    log(`eliminated ${pathCounts.length - MAX_TRANSITIVE_PATHS} paths with frequency <= ${pathCounts[MAX_TRANSITIVE_PATHS].count} / ${paths.length} `)
   }
 
-  // for now this is pretty simple, sort paths by the sequence of routes they use, and choose an example, eliminating the multiple-board/transfer-stops problem
-  paths.sort((p1, p2) => {
+  let allStopPatternStopSets = pathCounts
```

Comment:
So we go from `PathDescriptor`s which are `[stopId, pathId]` to `StopPatternStop`s which are `[stop, pattern, stop]` and when we accumulate them by segment I call that a set 😉 

...naming things, open to suggestions! 😬